src/test/fuzz/policy_estimator.cpp

```diff
@@ -58,7 +58,16 @@ FUZZ_TARGET(policy_estimator, .init = initialize_policy_estimator)
                         break;
                     }
                     const CTransaction tx{*mtx};
-                    mempool_entries.push_back(ConsumeTxMemPoolEntry(fuzzed_data_provider, tx));
+                    const CTxMemPoolEntry entry{ConsumeTxMemPoolEntry(fuzzed_data_provider, tx)};
+                    mempool_entries.emplace_back(
+                        entry.GetSharedTx(),
+                        entry.GetFee(),
+                        entry.GetTime().count(),
+                        entry.GetHeight(),
+                        entry.GetSequence(),
+                        entry.GetSpendsCoinbase(),
+                        entry.GetSigOpCost(),
+                        entry.GetLockPoints());
```

Comment:
Replacing a simple copy with listing all the details of the object seems cumbersome. If we want to just prevent accidental copies, how about doing something like this instead:

```c++
class CTxMemPoolEntry {
private:
    CTxMemPoolEntry(const CTxMemPoolEntry&) = default;
    struct ExplicitCopyTag { explicit ExplicitCopyTag() = default; };
public:
    static constexpr ExplicitCopyTag ExplicitCopy;
    CTxMemPoolEntry(ExplicitCopyTag /*unused*/, const CTxMemPoolEntry& entry) : CTxMemPoolEntry(entry) { }
    ...
};

    mempool_entries.emplace_back(CTxMemPoolEntry::ExplicitCopy, entry);
```

Reply:
This looks nice, I'll try to apply it.